packages/core/execution-context/core-execution-context-server-internal/src/execution_context_container.ts

```diff
@@ -51,13 +51,35 @@ function stringify(ctx: KibanaExecutionContext): string {
   return ctx.child ? `${stringifiedCtx};${stringify(ctx.child)}` : stringifiedCtx;
 }
 
+/** get the sum of this context's circuit breaker and all child circuit breakers of type `breaker` */
+function getCircuitBreaker(
+  ctx: KibanaExecutionContext,
+  breaker: keyof Required<KibanaExecutionContext>['circuitBreakers']
+): number {
+  return (
+    (ctx.circuitBreakers ? ctx.circuitBreakers[breaker] : 0) +
+    (ctx.child ? getCircuitBreaker(ctx.child, breaker) : 0)
+  );
+}
+
 export class ExecutionContextContainer implements IExecutionContextContainer {
-  readonly #context: Readonly<KibanaExecutionContext>;
+  readonly #context: KibanaExecutionContext;
```

Comment:
IMO, the `circuitBreakers` counters should be a property outside the `#context`.

It'll make it easier to handle parent-child relationships (why are we always incrementing the root property but summing parent + child if the child's counters can never increase?).

Also, it wouldn't open the ability to call `executionContext.withContext({ circuitBreakers: { requests: -Infinity } })` 😈